### ISSUE TITLE
Crash on Android 11

### ISSUE THREAD

**Issue by JonnyBurger** [square/okhttp#6569]
We received reports about the following crash for our React Native app in our crash reporting tool:

```
java.lang.IllegalStateException
Caused by java.lang.IllegalStateException: Expected Android API level 21+ but was 30
       at okhttp3.internal.platform.AndroidPlatform.buildIfSupported(AndroidPlatform.java:238)
       at okhttp3.internal.platform.Platform.findPlatform(Platform.java:202)
       at okhttp3.internal.platform.Platform.<clinit>(Platform.java:79)
       at okhttp3.internal.platform.Platform.get(Platform.java:85)
       at okhttp3.OkHttpClient.newSslSocketFactory(OkHttpClient.java:263)
       at okhttp3.OkHttpClient.<init>(OkHttpClient.java:229)
       at okhttp3.OkHttpClient$Builder.build(OkHttpClient.java:1015)
       at com.facebook.react.devsupport.DevServerHelper.<init>(DevServerHelper.java:132)
       at com.facebook.react.devsupport.DevSupportManagerBase.<init>(DevSupportManagerBase.java:171)
       at com.facebook.react.devsupport.DevSupportManagerImpl.<init>(DevSupportManagerImpl.java:72)
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at com.facebook.react.devsupport.DevSupportManagerFactory.create(DevSupportManagerFactory.java:80)
       at com.facebook.react.ReactInstanceManager.<init>(ReactInstanceManager.java:237)
       at com.facebook.react.ReactInstanceManagerBuilder.build(ReactInstanceManagerBuilder.java:280)
       at com.facebook.react.ReactNativeHost.createReactInstanceManager(ReactNativeHost.java:87)
       at com.facebook.react.ReactNativeHost.getReactInstanceManager(ReactNativeHost.java:39)
       at com.freyasavings.MainApplication.onCreate(MainApplication.java:74)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6712)
       at android.app.ActivityThread.access$1300(ActivityThread.java:237)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:223)
       at android.app.ActivityThread.main(ActivityThread.java:7656)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

The check is performed here: https://github.com/square/okhttp/blob/master/okhttp/src/main/kotlin/okhttp3/internal/platform/AndroidPlatform.kt#L154

I suspect this is a bug because the function is stating that the expectation was not met when it was. Running on Android 11.

**Comment by yschimke** [comment]
What version of OkHttp are you using?

This should be fixed in 3.14.7 and 4.9.1

https://github.com/square/okhttp/blob/edf477cb4e7b1d59c6e3e5ac00870619340ce990/docs/changelog_3x.md#version-3147

Similar error here https://github.com/square/okhttp/issues/6202

**Comment by JonnyBurger** [comment]
Thanks @yschimke!

I used a version before that, I forced it to use a newer version.
So I think it's resolved.

Looks like React Native has also already upgraded and this error will go away in the next release.

**Comment by AlfredAndroidTedmob** [comment]
So, there is no plan to fix that bug for branch 3.12.x, where we are not forced to update the minSDK?
